forte/data/base_pack.py

```diff
@@ -481,9 +483,19 @@ def get_single(self, entry_type: Union[str, Type[EntryType]]) -> EntryType:
         Returns:
             A single data entry.
         """
-        for a in self.get(entry_type):
-            return a
-
+        idx = -1
+        for idx, a in enumerate(self.get(entry_type)):
+            if idx == payload_index:
+                return a
```

Comment:
Don't understand this part, `get` is [returning a list of `Entry`](https://github.com/asyml/forte/pull/828/files#diff-08f2c9d027629fa160cbf4bbd3d8df70b58889cdd88632b3e369126c1c84d447R456) right? Why `payload_index` here?

Reply:
Also, enumerating and checking for index looks quite weird to me.

Reply:
I think this is `get_single()` which is intended to get a single entry. In the past, it could only get the first one, here I make it to get kth one. Maybe I should change parameter instead. The index checking is used when there is no entry at all or the requested index is not available.

Reply:
I feel like you are mixing up two different indices, what is the meaning of getting the kth item when k=payload index?

Reply:
We shouldn't support getting `k`th item, we could support get the first item for a certain payload, but that's not what is currently implemented here.

Reply:
Or maybe you just name the parameter wrong? Why is `payload_index` even appear here?